django/thunderstore/permissions/mixins.py

```diff
@@ -44,7 +56,7 @@ class VisibilityMixin(models.Model):
     @transaction.atomic
     def save(self, *args, **kwargs):
         if not self.pk and not self.visibility:
-            self.visibility = VisibilityFlags.objects.create_public()
+            self.visibility = VisibilityFlags.objects.create_private()
```

Comment:
This does not seem like a good idea to change, it'll mean any & every upload will de-facto be disabled until someone manually approves them, which is not a change we want to make this abruptly. (If you had something to account for this elsewhere, I've not reviewed it yet)

Reply:
This is accounted for by updating the visibility of every package listing and version when they are created. I made this change because in the event that something goes wrong, the default state isn't letting something that shouldn't be public be public.

Reply:
I'll get back to this once I've read a bit further and grasp what's going on in the bigger picture in regards to this, but for now I'd suggest keeping the mixin's default as-is and explicitly setting the flags where-ever new objects are being created where this should be applied to (given that the mixin is for any & every future object applicable to the visibility system as well, not just packages).

Keep in mind you can safely first create the object and then update the visibility flags as long as it's all in the same db transaction. The `package_upload.py` form processing logic would be a prime spot for anything to do with this for example.